dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceQueryParams.java

```diff
@@ -438,7 +406,10 @@ public TrackedEntityInstanceQueryParams addAttributes( List<QueryItem> attrs )
 
     public boolean hasFilterForEvents()
     {
-        return hasAssignedUsers() || isIncludeOnlyAssignedEvents() || isIncludeOnlyUnassignedEvents()
+        // in the new class)
+        return this.getAssignedUserQueryParam().hasAssignedUsers()
+            || AssignedUserSelectionMode.ANY == this.getAssignedUserQueryParam().getMode()
+            || AssignedUserSelectionMode.NONE == this.getAssignedUserQueryParam().getMode()
```

Comment:
Maybe here would be easier to understand with 

```suggestion
return AssignedUserSelectionMode.ALL != this.getAssignedUserQueryParam().getMode()
```

?

Reply:
Good find! That's my end goal 😋 yes 😄 

I did not yet add `AssignedUserSelectionMode.ALL`. I need to make another PR for EventSearchParams. Then I'll add AssignedUserSelectionMode.ALL. I think creating separate PRs will make it easier to follow.

Reply:
If doing this in steps is ok for you @enricocolasante can you please mark this as resolved 😄